src/Devices/EthernetServer_TC.cpp

```diff
@@ -420,3 +438,61 @@ int EthernetServer_TC::weekday(int year, int month, int day) {
   yy = year - adjustment;
   return (day + (13 * mm - 1) / 5 + yy + yy / 4 - yy / 100 + yy / 400) % 7;
 }
+
+void EthernetServer_TC::arbitraryPath() {
+  int next;
+  int count = 0;
+  while (bufferContentsSize < sizeof(buffer) - 1 && (next = client.read()) != -1) {  // Flawfinder: ignore
+    buffer[bufferContentsSize++] = (char)(next & 0xFF);
+    if (buffer[0] != '[') {
+      serial(F("Bad request body"));
+      sendBadBody();
+      return;
+    }
+
+    if (next == ',' || next == ']') {
+      buffer[bufferContentsSize - 1] = '\0';
+      SD_TC::instance()->writePhPoint(charArrToFloat(buffer + 1));
+      bufferContentsSize = 1;
+      if (next == ']') {
+        PHControl::instance()->setArbitrary();
+        static const char response[] PROGMEM =
```

Comment:
I like how `sendBadBody()` is its own method. I think that lines 459-466 should be in a similar function `sendCreated()`.